erpnext/hr/doctype/salary_structure/salary_structure.js

```diff
@@ -61,6 +61,11 @@ frappe.ui.form.on('Salary Structure', {
 				frappe.set_route('Form', 'Salary Structure Assignment', doc.name);
 			});
 		}
+		frm.set_df_property("is_tax_applicable", "read_only", 1, "Salary Detail", "earnings");
+		let fields_read_only = ["is_tax_applicable", "is_flexible_benefit", "variable_based_on_taxable_salary"];
```

Comment:
Why not used `read_only` property of docfield?

Reply:
its marked readonly, should we allow form customization to change it? line 64 is by mistake, removing it.